test/runtime-playwright/index.js

```diff
@@ -1,79 +1,38 @@
-import * as path from 'path';
+import virtual from '@rollup/plugin-virtual';
 import * as fs from 'fs';
-import * as http from 'http';
+import * as path from 'path';
 import { rollup } from 'rollup';
-import virtual from '@rollup/plugin-virtual';
-import puppeteer from 'puppeteer';
-
-import {
-	loadConfig,
-	loadSvelte,
-	mkdirp,
-	prettyPrintPuppeteerAssertionError,
-	retryAsync,
-	executeBrowserTest
-} from '../helpers';
-import { deepEqual } from 'assert';
-
-const page = `
-<body>
-	<main></main>
-	<script src='/bundle.js'></script>
-</body>
-`;
-
-let svelte;
-let server;
-let code;
-let browser;
+
+import { chromium } from '@playwright/test';
+import { deepStrictEqual } from 'assert';
+import { loadConfig, loadSvelte, mkdirp, prettyPrintBrowserAssertionError } from '../helpers';
 
 const internal = path.resolve('internal/index.mjs');
 const index = path.resolve('index.mjs');
 
-function create_server() {
-	return new Promise((fulfil, reject) => {
-		const server = http.createServer((req, res) => {
-			if (req.url === '/') {
-				res.end(page);
-			}
-
-			if (req.url === '/bundle.js') {
-				res.end(code);
-			}
-		});
-
-		server.on('error', reject);
+const assert = fs.readFileSync(`${__dirname}/assert.js`, 'utf-8');
 
-		server.listen('6789', () => {
-			fulfil(server);
-		});
-	});
-}
+describe('runtime (playwright)', function () {
+	this.timeout(10000);
 
-async function launchPuppeteer() {
-	return await retryAsync(() => puppeteer.launch());
-}
+	let svelte;
+	let browser;
 
-const assert = fs.readFileSync(`${__dirname}/assert.js`, 'utf-8');
-
-describe('runtime (puppeteer)', () => {
 	before(async () => {
 		svelte = loadSvelte(false);
-		console.log('[runtime-puppeteer] Loaded Svelte');
-		server = await create_server();
-		console.log('[runtime-puppeteer] Started server');
-		browser = await launchPuppeteer();
-		console.log('[runtime-puppeteer] Launched puppeteer browser');
+		console.log('[runtime-playwright] Loaded Svelte');
+
+		browser = await chromium.launch();
```

Comment:
Are you going for a minimal diff change to see it all passing first and then refactor for more idiomatic playwright use later? because launching browsers manually in before is not what you typically do anymore.

Reply:
> Are you going for a minimal diff change to see it all passing first and then refactor for more idiomatic playwright use later?

Yesn't. I did want to keep things as close to the puppeteer implementation for now, just to make sure no tests get missed. Although, I don't know what a more idiomatic way would be either apart from playwright's own test runner.

Are you referring to reusing the server globally like in the vite repo?

Reply:
i was refering to using playwrights testrunner, yes